src/roles/common/tasks/check-predeploy-prereq.yml

```diff
@@ -20,8 +20,19 @@
 
 - block:
 
-  - block:
+  - name: Check that image files are not missing
+    set_fact:
+      is_file_missing: "{{ image_path | default ('MISSING') == 'MISSING' or vrs_package_path | default ('MISSING')  == 'MISSING' or dkms_package_path | default ('MISSING') == 'MISSING' }}"
```

Comment:
You will need to remove the vrs_package_path and dkms_package_path from here.  These only apply when component_type == 'vrs'.  Also, you are missing the "not vsd_inplace_upgrade" check at line 61.  Instead of trying to capture all of the logic here, I'd recommend also doing the md5 delete in the component_type == 'vrs' block below like you are doing here.  Don't forget to check the dkms_install check at line 57 when you add that code.